### ISSUE TITLE
opencv (3.x) not installing contrib packages

### ISSUE THREAD

**Issue by turowicz** [Homebrew/homebrew-core#30679]
Homebrew/science tap was merged to core but the opencv formula lost the option `--with-contrib` and `--with-python3` that are crucial to making majority of opencv-based projects to work.

**Comment by commitay** [comment]
It has both on by default.

https://github.com/Homebrew/homebrew-core/blob/06eae891074e0b879e742bf54560f89a11f183df/Formula/opencv.rb#L77

https://github.com/Homebrew/homebrew-core/blob/06eae891074e0b879e742bf54560f89a11f183df/Formula/opencv.rb#L63

**Comment by turowicz** [comment]
@commitay it's not installing the contrib libs though.

**Comment by turowicz** [comment]
@commitay let me rephrase, the contrib headers are not listed in the opencv include file after installation.

**Comment by ilovezfs** [comment]
If something beyond setting `-DOPENCV_EXTRA_MODULES_PATH=#{buildpath}/opencv_contrib/modules` is needed, we can review PRs with potential fixes. You may need to ask upstream for help (https://github.com/opencv/opencv/issues) figuring out what would need to be changed.

**Comment by turowicz** [comment]
Sorry I won't be able to help with this as I'm not familiar with formulas. Installing opencv manually with make does the job. I don't know why it doesn't work through the brew script.

**Comment by ilovezfs** [comment]
What is the specific difference you're noticing?

**Comment by turowicz** [comment]
Contrib l.h files aren't included in the `/usr/local/include/` folder (OSX in my case)

**Comment by turowicz** [comment]
edit: I've followed this simple guide. Basically clone and run make.

https://www.webuildinternet.com/2016/05/30/installing-opencv-contrib-on-osx/

**Comment by helvalius** [comment]
I stumbled upon the same issue. A real pitty that the opencv_contrib is missing. I would need to invest some time to see if I can do a pull request, but haven't done any formula yet.

**Comment by SMillerDev** [comment]
Why though?

**Comment by turowicz** [comment]
@commitay could you reopen this issue as other people also experience the problem?

**Comment by SMillerDev** [comment]
> If something beyond setting -DOPENCV_EXTRA_MODULES_PATH=#{buildpath}/opencv_contrib/modules is needed, we can review PRs with potential fixes. You may need to ask upstream for help (https://github.com/opencv/opencv/issues) figuring out what would need to be changed.

These documents might help you get started contributing to homebrew:
https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
https://docs.brew.sh/Formula-Cookbook
